core/modules/utils/dom/dom.js

```diff
@@ -311,6 +311,48 @@ exports.getLocationPath = function() {
 	return window.location.toString().split("#")[0];
 };
 
+exports.copyEventProperties = function(event) {
+	var seen = new Set();
+  
+	function isDOMElement(value) {
+	  return value instanceof Node || value instanceof Window;
+	}
+  
+	function safeCopy(obj) {
+		//skip ciruclar references
+		if(seen.has(obj)) {
+			return "[Circular reference]";
+		}
+		//skip functions
+		if(typeof obj !== "object" || obj === null) {
+			return obj;
+		}
+		//skip DOM elements
+		if(isDOMElement(obj)) {
+			return "[DOM Element]";
+		}
+		//copy each element of the array
+		if(Array.isArray(obj)) {
+			return obj.map(safeCopy);
+		}
+
+		seen.add(obj);
+		var copy = {}, key;
+		for(key in obj) {
+			try{
+				copy[key] = safeCopy(obj[key]);
+			} catch(e) {
```

Comment:
When does this happen?  `safeCopy` shouldn't throw, should it?

Reply:
Hi @CrossEye, thank you for taking a look. Some event properties have the possiblity to throw exceptions when accessed (`event.target.contentWindow` for cross-origin iframes for example), and in particular I have run across libraries that implement custom events in this manner as well. So the `try/catch` is in the interest of robustness.